### PR TITLE
MessageBus: Rest Interface using aiohttp

### PR DESCRIPTION
**Pull request by selvakumaar5496**
Signed-off-by: Selvakumar <selva.k.nambi@seagate.com>

### CODE REVIEW

File: py-utils/src/utils/message_bus/messagebus-rest.service [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=cortx rest server
+After=network.target
+
+[Service]
+User=root
+WorkingDirectory=/usr/lib/python3.6/site-packages/cortx/utils
+ExecStart=/usr/lib/python3.6/site-packages/cortx/utils/message_bus/message_bus_server
```
Comment (ujjwalpl) [inline py-utils/src/utils/message_bus/messagebus-rest.service:8]: Path should be /opt/seagate/cortx/utils/message_bus/message_bus_server. Add this path in setup.py

File: py-utils/src/utils/message_bus/message_bus_server [hunk 2]
```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
```
Comment (ujjwalpl) [inline py-utils/src/utils/message_bus/message_bus_server:1]: Where is the test code ?

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/message_bus/message_bus_server [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    """ Rest interface of message bus """
+
+    def __init__(self):
+        app = web.Application()
+        app.add_routes(routes)
+        web.run_app(app)
+
+    @staticmethod
+    @routes.get('/MessageBus/{message_type}')
+    @routes.put('/MessageBus/{message_type}')
```
[the remark this answers is not in the document]
Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_bus_server:35]: Ip and port allocated as per the Spec document

File: py-utils/src/utils/message_bus/message_bus_server [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+class MessageBusRestServer:
+    """ Rest interface of message bus """
+
+    def __init__(self):
+        app = web.Application()
+        app.add_routes(routes)
+        web.run_app(app)
+
+    @staticmethod
+    @routes.get('/MessageBus/{message_type}')
```
[the remark this answers is not in the document]
Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_bus_server:34]: separate RestServer class is created and MessageBusRestServer is a subclass of it. Any new rest server implementation can use RestServer interface

File: py-utils/src/utils/message_bus/message_bus_server (hunk 2 above)
Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_bus_server:1]: Test code is added

**Comment by selvakumaar5496** [comment]
created a new PR for this, hence closing this.

**Pull request closed without merging**